src/recon.rs

```diff
@@ -263,21 +265,25 @@ impl_MergeInt!(u32, u16);
 impl_MergeInt!(u64, u32);
 impl_MergeInt!(u128, u64);
 
-#[inline]
-fn get_skip_ctx(
-    t_dim: &TxfmInfo,
+#[inline(always)]
+fn get_skip_ctx<const txsize_discr: usize>(
     bs: BlockSize,
     a: &[u8],
     l: &[u8],
     chroma: bool,
     layout: Rav1dPixelLayout,
 ) -> InRange<u8, 0, { 13 - 1 }> {
+    macro_rules! tdim {
+        () => {
+            dav1d_txfm_dimension::<txsize_discr>()
+        };
+    }
```

Comment:
I take it making `tdim` a local here did not optimize well?

Reply:
I actually haven't tried on the most recent version of the code: I initially only wanted to show the results of just the const generics with this "more readable"/smaller diff ergonomics change. I didn't really expect we could push it further.

I will check though.

Reply:
It looks like it, but it may be possible to do better. More details in https://github.com/memorysafety/rav1d/pull/1325#discussion_r1683339486